mongodbatlas/logs.go

```diff
@@ -0,0 +1,72 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+)
+
+const logsPath = "groups/%s/clusters/%s/logs/%s"
+
+// LogsService is an interface for interfacing with the Logs
+// endpoints of the MongoDB Atlas API.
+// See more: https://docs.atlas.mongodb.com/reference/api/logs/
+type LogsService interface {
+	Get(context.Context, string, string, string, *LogsListOptions) (*Response, error)
+}
+
+// LogsServiceOp handles communication with the Logs related methods of the
+// MongoDB Atlas API
+type LogsServiceOp struct {
+	Client GZipRequestDoer
+}
+
+// EventListOptions specifies the optional parameters to the Event List methods.
+type LogsListOptions struct {
+	StartDate string `url:"startDate,omitempty"`
+	EndDate   string `url:"endDate,omitempty"`
+}
+
+// Get gets a compressed (.gz) log file that contains a range of log messages for a particular host in an Atlas cluster.
+// See more: https://docs.atlas.mongodb.com/reference/api/logs/
+func (s *LogsServiceOp) Get(ctx context.Context, groupID string, hostName string, logName string, opts *LogsListOptions) (*Response, error) {
+	if groupID == "" {
+		return nil, NewArgError("groupID", "must be set")
+	}
+
+	if hostName == "" {
+		return nil, NewArgError("hostName", "must be set")
+	}
+
+	if logName == "" {
+		return nil, NewArgError("logName", "must be set")
+	}
+
+	basePath := fmt.Sprintf(logsPath, groupID, hostName, logName)
+
+	// Add query params
+	path, err := setListOptions(basePath, opts)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := s.Client.NewGZipRequest(ctx, http.MethodGet, path)
+	if err != nil {
+		return nil, err
+	}
+
+	out, err := os.Create(logName)
```

Comment:
Right now this method is creating the output file but I think we should pass it the file, an `io.Writer` probably or accept the name of the output file I don't know which one is better

Reply:
why do you think so? I do not see any advantages in changing it. It seems really clean in this way.

Reply:
the main reason to change it is that users don't have control over the name file which may or may not be desirable, that's why at least passing the output file name should be enough
on the other hand the problem with using `os.Create(logName)` directly is that we depend on a particular implementation of `io.Writer` which is kind of bad as not all the users of this client may actually have access to a file system but may still have a way to write files
I think an ideal world we could have this `Get` method with an `io.Writer` param, and then we can create a `Download` method which does this assuming `os.Create(logName)`

Reply:
Do you want me to make this change (`io.Writer` as param)? I am not sure how to implement the download method though

Reply:
yes, let's create the two method in this PR as getting this merge takes some time so we don't have to revisit this in case we need it.
on how to implement we should split the current method in two, get should not do the os.create but directly pass the parma to the Do method and thus makeing this method a bit simpler
Then the download method should take the logName and create a file, just like we are doing here now, and then with that new file call the simpler Get method, does that make sense?

Reply:
Just to be sure that we are on the same page, I am going to add the writer to the `Get` but I will not remove the logName( I may change its name) because according to the  documentation it allows to retrieve different type of log file:
- mongodb.gz
- mongos.gz
- mongodb-audit-log.gz
- mongos-audit-log.gz